### DISCUSSION TITLE
Custom `baseUrl` input?

### DISCUSSION THREAD

**Discussion by BrutuZ** [Animeboynz/Mihon-Backup-Viewer#84] [Polls]
I wondered if we should have a way for users to manually set a `baseUrl` (and "`chapterBaseUrl`") for their ~~extensions~~ sources under settings.

It would be exclusively for the page to form full URLs to entries on the details modal as most are saved as relative paths.

**Comment by Animeboynz** [comment]
Well I feel like if we wanted to do this, there are a few ways to accomplish it.
- The user enters the base URL of each source manually (Would would drive me mad if it reset every time I load a backup)
- We add a custom upload box in the setting for the user to upload a file which contains the Base URLs.
  - We can provide this file to the user but I'm not sure about the legality of that.

**Reply by Animeboynz** [reply]
I wish extensions would just be consistent and always have full URLs for chapters😒

**Comment by BrutuZ** [comment]
I'm not too worried about the "how", a separate modal with a couple text fields for each source or a textarea to be filled with a JSON would work just fine.
I'm asking if it's something "only I would use" or it might be useful in general.

Full URLs are actually bad. Whenever a source changes their hostname for whatever reason, all the history is gone after the extension gets updated, and if the main entry uses that too then it's even worse as you have to migrate it.

**Reply by Animeboynz** [reply]
I guess that makes sense. 
I think it would be useful in my opinion.

**Reply by Animeboynz** [reply]
I think my favorite way of implementing this would be slightly similar to Mihon.
In settings or wherever it could have a option to `Add Extension Repo` and the user provides the ling to their `index.min.json`.
and that should contain the baseUrl for each extension.

And if [Mihon #762](https://github.com/mihonapp/mihon/issues/762) gets added it can be automated too.

**Reply by BrutuZ** [reply]
That would greatly overcomplicate for the user as they'd have to host that somewhere. Just fill it right there and save to LocalStorage.

~~Weird that issue is open, it's already in the backup, inside `backupPreferences` with the key `extensions_repo`~~ Probably the usual case of it being a SY Previews feature without me realizing it xD

**Reply by Animeboynz** [reply]
Bruh your right. It does get backed up and even restored in Mihon as well.

**Reply by BrutuZ** [reply]
Stable vs Preview moment
